spring-cloud-aws-sqs/src/main/java/io/awspring/cloud/sqs/operations/SqsTemplate.java

```diff
@@ -616,6 +616,15 @@ private ReceiveMessageRequest doCreateReceiveMessageRequest(Duration pollTimeout
 		return builder.build();
 	}
 
+	// Convert a long value to an int. Values larger than Integer.MAX_VALUE are set to Integer.MAX_VALUE
+	private int toInt(long longValue) {
```

Comment:
I didn't want to use `Math.toIntExact()` as it will throw if the long value is out of the range and I thought pinning to the max integer would be better behavior. If we think it's unlikely enough that someone would set a timeout that high (over 63 years), then I'm happy to switch to the built in.

I didn't bother pinning too small of negative values as I don't think anyone would intentionally set a negative timeout.

Reply:
We can go with 2 approaches.
1. Yours by rounding to maximum integer there is, which is totally valid.

2. Or follow AWS SDK approach where we would (validate) throw exception with stating that value is above limit.

I personally would maybe want to go with approach number 2 just to be aligned, but both approaches are totally valid.
But tbh we are now nitpicking since no one will probably want or will set timeout that big. So for me this is fine!